### PR TITLE
ContentFiles for edx/ocw learning resources

### LINKED ISSUE #18: Models, serializers, and API views for ContentFiles
Write the necessary models, serializers, and views to represent course content files in an API.

#### Designs and Mockups

See docs mentioned in #4 and and #11

#### Acceptance Criteria:

- [ ] Working API for content files

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist

- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #18 

#### What's this PR do?
- Adds `ContentFile` model, serializers, views, urls including 3 endpoints:
  ```
  /api/v1/contentfiles/
  /api/v1/learning_resources/<resource_id>/contentfiles/
  /api/v1/courses/<resource_id>/contentfiles/
  ```  
- Moves the `backpopulate_xpro_files` management command from `course_catalog` to `learning_resources`
- Related: https://github.com/mitodl/salt-ops/pull/1583

#### How should this be manually tested?
Make sure the following are populated in your .env:
```
XPRO_LEARNING_COURSE_BUCKET_NAME=<same as heroku open-discussions-rc value>
XPRO_CATALOG_API_URL=https://xpro.mit.edu/api/programs/
XPRO_COURSES_API_URL=https://xpro.mit.edu/api/courses/
AWS_ACCESS_KEY_ID=<same as heroku mit-open-rc value>
AWS_SECRET_ACCESS_KEY=<same as heroku mit-open-rc value>
```

Run the following:
```
manage.py backpopulate_xpro_data
manage.py backpopulate_xpro_files
```
Both should complete successfully.

Try the following urls, they should all work and return results:
```
http://localhost:8063/api/v1/contentfiles/
http://localhost:8063/api/v1/contentfiles/?run=<some_run__id>/
http://localhost:8063/api/v1/learning_resources/<xpro_resource_id>/
http://localhost:8063/api/v1/learning_resources/<xpro_resource_id>/?run=<some_run__id>/
http://localhost:8063/api/v1/courses/<xpro_resource_id>/
http://localhost:8063/api/v1/courses/<xpro_resource_id>/?run=<some_run__id>/
```

### CODE REVIEW

File: learning_resources/serializers.py [hunk 1]
```diff
@@ -375,5 +380,74 @@ class LearningPathRelationshipSerializer(LearningResourceRelationshipSerializer)
     """Specialized serializer for a LearningPath relationship"""
 
     relation_type = serializers.HiddenField(
-        default=LearningResourceRelationTypes.LEARNING_PATH_ITEMS.value
+        default=constants.LearningResourceRelationTypes.LEARNING_PATH_ITEMS.value
+    )
+
+
+class ContentFileSerializer(serializers.ModelSerializer):
```
Comment (mbertrand) [inline learning_resources/serializers.py:387]: There are some fields not included here that will be required for opensearch (namely, `resource_relations` that indicates the parent-child relationship between the parent course and the contentfile). My thinking is that there will still be a `search.serializers.OSContentFileSerializer` that could inherit from this serializer and add that field.  Let me know if you'd prefer all the fields required for opensearch to be defined here.

**Review by abeglova — CHANGES_REQUESTED** [review]
One comment, otherwise lgtm

**Review by abeglova — APPROVED** [review]
LGTM

**Comment by abeglova** [comment]
Sorry, I thought I already approved this...

**Pull request merged**